### ISSUE TITLE
Contexts from kubectl being used, instead of the configured provider settings.

### ISSUE THREAD

**Issue by drzero42** [terraform-providers/terraform-provider-kubernetes#92]
### Terraform Version
- Terraform v0.11.0
- provider.kubernetes v1.0.0

### Affected Resource(s)
- kubernetes_config_map
- kubernetes_secret

I have setup a kubernetes cluster with the google_container_cluster resource and use the outputs to configure the kubernetes provider to add some configmaps and secrets to the newly created cluster. I have several different contexts setup for my kubectl, which I switch between. If something about the cluster has changed, any "terraform plan" runs will not use the provided information to connect to the cluster, to refresh the state, but instead uses current context in kubectl, to connect, which means that it thinks the configmaps and secrets in the cluster are wrong, because it is talking to the wrong cluster. It should not use the current-context from ~/.kube/config, but instead connect to the endpoint the provider is configured to use. Let me know if you need additional information. I will happily provide whatever you need.

**Comment by sl1pm4t** [comment]
Hey @drzero42 can you paste in your `kubernetes_provider` definition?

**Comment by drzero42** [comment]
@sl1pm4t Sure thing!
It is packaged as a module. I created a gist with everything in the module: https://gist.github.com/drzero42/31a1409a30a2d6288f6c655f4947ed15

**Comment by sl1pm4t** [comment]
Ok @drzero42  the provider definition looks fine to me. I also have many kubectl contexts configured, but I don't have any issues with the kubernetes providers using the wrong config.

Some things to try / check:

* Set the provider `load_config_file` value to false.

* Check if you have any of the following environment variables set that start with `KUBE_`, e.g.
  * `KUBE_HOST`
  * `KUBE_CTX`
  * `KUBE_CTX_AUTH_INFO`
  * `KUBE_CTX_CLUSTER`

**Comment by drzero42** [comment]
Okay, `load_config_file = false` makes a difference, which has given me a clue as to what is actually going on. I think this might be related to the way Terraform works. My specific situation is that I have a TF state with two modules, one that sets up a GKE cluster (with the Google provider), and one that adds some configmaps and secrets to it (with the Kubernetes provider). I did some changes to the `google_container_cluster` resources, which means the current GKE resources will be destroyed and new ones created. As the outputs from the container cluster module are used as inputs to the kubernetes module (and thus to configure the provider), and these are now marked as changing, the kubernetes provider falls back to using values from ~/.kube/config. If I add `load_config_file = false` it fails altogether, as it then tries default values which seems to be, amongst other things, 127.0.0.1:80 as the endpoint. I think all I can do here is delete the Kubernetes (configmap/secret) resources from the state, as they will be gone when the cluster gets destroyed anyways, and then everything will be fine. I am guessing some sort of cascading taint/dependency handling in Terraform would be needed for this to be automatic. For now, it requires operator knowledge and intervention :)
Thank you for your help. Closing ticket.
